tests/testClassification.py

```diff
@@ -75,6 +75,116 @@ def testAlgorithm(self):
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertTrue(source.get("base_ClassificationExtendedness_flag"))
+
+        #  Test modelFlux NAN case
+        source = measCat.addNew()
+        source.set("base_PsfFlux_flux", 100)
+        source.set("base_PsfFlux_fluxSigma", 1)
+        source.set("base_SincFlux_fluxSigma", 2)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        source.set("base_SincFlux_flag", True)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertTrue(source.get("base_ClassificationExtendedness_flag"))
+
+        #  Test psfFlux NAN case
+        source = measCat.addNew()
+        source.set("base_PsfFlux_fluxSigma", 1)
+        source.set("base_SincFlux_flux", 200)
+        source.set("base_SincFlux_fluxSigma", 2)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        source.set("base_SincFlux_flag", True)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertTrue(source.get("base_ClassificationExtendedness_flag"))
+
+        #  Test modelFluxErr NAN case when modelErrFactor is zero and non-zero
+        sfm_config.plugins["base_ClassificationExtendedness"].modelErrFactor = 0.
+        source = measCat.addNew()
+        source.set("base_PsfFlux_flux", 100)
+        source.set("base_PsfFlux_fluxSigma", 1)
+        source.set("base_SincFlux_flux", 200)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertFalse(source.get("base_ClassificationExtendedness_flag"))
+
+        sfm_config.plugins["base_ClassificationExtendedness"].modelErrFactor = 1.
+        source = measCat.addNew()
+        source.set("base_PsfFlux_flux", 100)
+        source.set("base_PsfFlux_fluxSigma", 1)
+        source.set("base_SincFlux_flux", 200)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertTrue(source.get("base_ClassificationExtendedness_flag"))
+
+        #  Test psfFluxErr NAN case when psfErrFactor is zero and non-zero
+        sfm_config.plugins["base_ClassificationExtendedness"].psfErrFactor = 0.
+        source = measCat.addNew()
+        source.set("base_PsfFlux_flux", 100)
+        source.set("base_SincFlux_fluxSigma", 1)
+        source.set("base_SincFlux_flux", 200)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertFalse(source.get("base_ClassificationExtendedness_flag"))
+
+        sfm_config.plugins["base_ClassificationExtendedness"].psfErrFactor = 1.
+        source = measCat.addNew()
+        source.set("base_PsfFlux_flux", 100)
+        source.set("base_SincFlux_fluxSigma", 1)
+        source.set("base_SincFlux_flux", 200)
+        task.plugins["base_ClassificationExtendedness"].measure(source, exp)
+        self.assertTrue(source.get("base_ClassificationExtendedness_flag"))
```

Comment:
You could write the same tests with a lot less code by defining a method which takes the source parameters and expected flag value as arguments and then calling it repeatedly.